stdlib/public/TensorFlow/OpaqueHandles.swift

```diff
@@ -16,61 +16,117 @@
 
 import CTensorFlow
 
-/// `ResourceHandle` is the type used by ops and the `#tfop()` syntax to
-/// represent TensorFlow "resource" values.  It exists only to represent edges
-/// in TensorFlow graphs, and has no host-side representation (and thus no
-/// methods).
-public final class ResourceHandle {
-  private init() {
-    fatalError("ResourceHandle is a marker type that can never be created")
-  }
-}
-
-/// `VariantHandle` is the type used by ops and the `#tfop()` syntax to
-/// represent TensorFlow "variant" values.
-/// TODO: rename this source file as this is no longer "opaque".
-public final class VariantHandle {
-  public let cTensorHandle: CTensorHandle
-
+/// A class that implements common functionality for resource and variant handles.
+public class ResourceVariantHandleBase {
```

Comment:
Please do not introduce XXXBase-like data structures. This is an anti-pattern. If you really need to, it needs to be a protocol. You could call it `OpaqueHandleProtocol`.

But in any case, this refactoring is really unnecessary and it is making the code much more complicated. With the `isResource` flag, this is semantically a base type for only two possible conforming types or subtypes but does not statically enforce it.